deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -210,17 +213,23 @@ private void ingestChildren(ContentContainerObject destObj, Resource parentResc)
 
                 // Ingest the child according to its object type
                 if (childResc.hasProperty(RDF.type, Cdr.Folder)) {
-                    ingestFolder(destObj, parentResc, childResc);
+                    ingestFolder(destObj, parentResc, childResc, groupSet);
                 } else if (childResc.hasProperty(RDF.type, Cdr.Work)) {
-                    ingestWork(destObj, parentResc, childResc);
-                } else if (childResc.hasProperty(RDF.type, Cdr.FileObject)) {
+                    ingestWork(destObj, parentResc, childResc, groupSet);
+                } else if (childResc.hasProperty(RDF.type, Cdr.FileObject)
+                        || !childResc.hasProperty(RDF.type)) {
```

Comment:
i wonder whether this condition should be moved higher up in the block, so that a bunch of unnecessary calls of childResc.hasProperty(RDF.type) aren't made